lib/explorer/series.ex

```diff
@@ -92,11 +94,13 @@ defmodule Explorer.Series do
           | :integer
           | :string
           | list_dtype
+          | struct_dtype
 
   @type time_unit :: :nanosecond | :microsecond | :millisecond
   @type datetime_dtype :: {:datetime, time_unit}
   @type duration_dtype :: {:duration, time_unit}
   @type list_dtype :: {:list, dtype()}
+  @type struct_dtype :: {:struct, %{String.t() => dtype()}}
```

Comment:
Hi! I really like this PR and think this missing functionality would be very helpful.

That said, I'm hesitant about the term "struct" since as you say it's overloaded. What do you think about `DF.from_rows`? Then it would pair with [`DF.to_rows`](https://hexdocs.pm/explorer/Explorer.DataFrame.html#to_rows/2) just like `Series.from_list`/`Series.to_list`.

Reply:
So this is not importing data from Elixir structs, but literally having a struct column inside Explorer. You could even have lists of structs. :)

Reply:
1. Whoops! Thank you, I was skimming 😅 
2. I still worry that "struct" is overloaded ("map" may be better?), but now less so. It's sorta like how we have `{:f, 32}` even though there's no real equivalent in Elixir.

Reply:
Yeah, I was a bit iffy about naming it `struct` as well because of the pre-existing meaning within Elixir, but also naming it the more Elixir-appropriate `map` would mean people used to Polars would have to keep this "translation" in mind when using Explorer.

Feels like a "pick your poison" type of choice 😅

Reply:
I think we could solve the naming issue by documenting it (like I mentioned in my other comment). Since we are following Arrow data structs, we could just document that. WDYT?

Reply:
@philss Yeah if struct aligns with polars and arrow then it's the right call 👍 And documenting the name in case of potential confusion is def the way to go.

Reply:
I've added the extra bits to the docs here: e565494

WDYT?